### ISSUE TITLE
[SPIKE] [Test results] Date picker component reads incorrectly and does not provide correct instructions for usage.

### ISSUE THREAD

**Issue by johanna-skylight** [CDCgov/prime-simplereport#5862]
## Area or Component

Both date picker components used to set the start and end date in the results filter.

## Quick Summary

The date picker components does not get read by the screen reader correctly. Also the instructions  provided by the screen reader are incorrect making a non sighted user unable to use this component with keyboard alone.

### How to Replicate
Go to the results page and enable voice over. Notice that the date values are not properly read, specially when there is no value set. And then attempt to fill in the date by following the instructions of the screen reader.

## Accessibility Violation

[There is no way to perform the function using only the keyboard](https://docs.deque.com/issue-help/1.0.0/en/keyboard-inaccessible).

## Action Requested

~Fix the component so the values displayed in it are properly read by the screen reader (specially the mm/dd/yyyy state). Also make sure that the instructions presented on how to use the component are clear. You should be able to set a date in this component by following the instructions alone.~

Spike out whether this is an issue. Do some research on whether the `input type="date"` is accessible and/or if there are ways to make it more accessible. Perhaps try using a different screen reader in Chrome. Perhaps use the screen reader and try to reproduce this with the same type of date picker on other websites (e.g. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date)

Outcome of the spike is that we have a determination on whether to make a fix here and if so, a suggested plan forward with a ticket created.

**Comment by emyl3** [comment]
**Research resources**
- [Is input type=”date” ready for use in accessible websites?](https://www.hassellinclusion.com/blog/input-type-date-ready-for-use/#detailed-test-findings) 
   - published in 2019 (changes might have happened since -- ex: https://bugs.webkit.org/show_bug.cgi?id=119175)
   - pretty thorough test if `input type="date"` is accessible using various browsers and assistive technologies
- [gov.uk design system resource about 3 input dates](https://design-system.service.gov.uk/components/date-input/) (like how our pxp date input looks) 
   - the 3 input dates wouldn't work for the test results page because it will be helpful for the user to have a calendar visible to select a date
- [Someone's gist about how "Safari's date-picker is the cause of 1/3 of our customer support issues"](https://gist.github.com/RobertAKARobin/850a408e04d5414e67d308a2b5847378) 😅 

accessible date pickers
- [article 1](https://www.webaxe.org/accessible-date-pickers/)
- [article 2](https://www.hassellinclusion.com/blog/collecting-dates-accessible/)

**Conclusion**
- The users accessing out test results page are primarily desktop users.
- Screen reader support for

Immediate recommendations
- leave `<input type="date">` as-is for now
- add [`role="search"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/search_role) to this [`<div>`](https://github.com/CDCgov/prime-simplereport/blob/41dc26e99d99af52118837fd5551e26fbffb8efd/frontend/src/app/testResults/TestResultsList.tsx#L465) 
- remove the uneccessary `aria-label`s on the start and end date pickers: [here](https://github.com/CDCgov/prime-simplereport/blob/41dc26e99d99af52118837fd5551e26fbffb8efd/frontend/src/app/testResults/TestResultsList.tsx#L499) and [here](https://github.com/CDCgov/prime-simplereport/blob/41dc26e99d99af52118837fd5551e26fbffb8efd/frontend/src/app/testResults/TestResultsList.tsx#L513)

Future considerations
- if we want to support this for mobile and more combinations of browsers and screen readers, we may want to look into implementing our own version of [Deque's datepicker](https://dequeuniversity.com/library/aria/date-picker)

**Comment by emyl3** [comment]
ticket created: #6021

**Comment by emyl3** [comment]
added a note here: https://github.com/CDCgov/prime-simplereport/issues/5493
